core/shared/src/main/scala/fs2/concurrent/Signal.scala

```diff
@@ -73,10 +74,16 @@ object Signal extends SignalInstances {
   }
 
   implicit class BooleanSignalOps[F[_]](val self: Signal[F, Boolean]) extends AnyVal {
-    def interrupt[A](
-        s: Stream[F, A]
-    )(implicit F: Concurrent[F]): Stream[F, A] =
+
+    /** Interrupts the supplied `Stream` when this `Signal` is `true`.
+      */
+    def interrupt[A](s: Stream[F, A])(implicit F: Concurrent[F]): Stream[F, A] =
       s.interruptWhen(self)
+
+    /** Predicate the supplied effect `f` on this `Signal` being `true`.
+      */
+    def predicate(f: F[_])(implicit F: Applicative[F] with FlatMap[F]): F[Unit] =
```

Comment:
```suggestion
    def predicate(f: F[_]: Monad): F[Unit] =
```

Reply:
Hmm, that form doesn't compile for me, but this form does:
`def predicate(f: F[_])(implicit F: Monad[F]): F[Unit]`
Is that ok?

Reply:
Whoops, yep

Reply:
Great; committed.